lib/go/thrift/middleware.go

```diff
@@ -107,3 +110,76 @@ func WrapClient(client TClient, middlewares ...ClientMiddleware) TClient {
 	}
 	return client
 }
+
+// For a endpoint defined in thrift IDL like this:
+//
+//     service MyService {
+//       FooResponse foo(1: FooRequest request) throws (
+//         1: Exception1 error1,
+//         2: Exception2 error2,
+//       )
+//     }
+//
+// The thrift compiler generated go code for the result TStruct would be like:
+//
+//     type MyServiceFooResult struct {
+//       Success *FooResponse `thrift:"success,0" db:"success" json:"success,omitempty"`
+//       Error1 *Exception1 `thrift:"error1,1" db:"error1" json:"error1,omitempty"`
+//       Error2 *Exception2 `thrift:"error2,2" db:"error2" json:"error2,omitempty"`
+//     }
+func getClientError(result TStruct, err error) error {
```

Comment:
@dcelasun I'm also debating whether to export this as a public API, and if yes, what's a good name for it.

Reply:
I think it's fine to make it public, but maybe move it to `exception.go` and call it `GetClientException`?

Reply:
What do you think of `ExtractExceptionFromResult`?

Reply:
Sounds good!